docs/distribution/README.md

```diff
@@ -0,0 +1,222 @@
+# OCI Distribution
+
+We introduce new REST APIs in the registry to support storing signature objects together with the target artifacts, and retrieving them for verification.
+
+## Table of contents
+1. [Sample APIs for retrieving signatures](#Sample-APIs-for-retrieving-signatures)
+2. [Sample APIs for storing signatures](#Sample-APIs-for-storing-signatures)
+3. [Implementation](#Implementation)
+
+## Sample APIs for retrieving signatures
+
+The list of signatures for an artifact can be retrieved from the registry.
+
+### _Get a list of paginated signatures by tag_
+
+### Request
+`GET http://localhost:5000/v2/hello-world/manifests/v1.0/signatures/`
+
+### Response
+
+```json
+{
+    "tag": "v1.0",
+    "digest": "sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+    "@nextLink": "{opaqueUrl}",
+    "signatures": [
+        "sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+        "sha256:3335d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc88d"
+    ]
+}
+```
+
+### _Get a list of paginated signatures by digest_
+
+### Request
+`GET http://localhost:5000/v2/hello-world/manifests/sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/`
+
+### Response
+
+```json
+{
+    "digest": "sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+    "@nextLink": "{opaqueUrl}",
+    "signatures": [
+        "sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+        "sha256:3335d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc88d"
+    ]
+}
+```
+
+## Sample APIs for storing signatures
+
+### _Put an OCI index linking a signature to a collection of manifests by tag_
+
+### Request
+`PUT https://localhost:5000/v2/hello-world/manifests/signature-1`
+```json
+{
+  "schemaVersion": 2.1,
+  "mediaType": "application/vnd.oci.image.index.v2+json",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+jwt",
+    "digest": "sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+    "size": 1906
+  },
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "size": 7023,
+      "platform": {
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+```
+
+### _Put an OCI index linking a signature to a collection of manifests by digest_

```

Comment:
Is this a new PUT operation? We agreed to use the existing index put APIs, so I'm confused if this is a new proposal.
I'm also a little skeptical about storing "signature" layers, as opposed to simply storing an index that has a reference to an artifact.
This suggests signature objects are very special. 
I realize the get signatures api is unique, but we're trying to minimize the differences.

I'm also not super familiar with how storage is handled. Is the storage structure below unique to ACR, and every registry implements their own? Or, is this standard across all registries?

Reply:
I'm all for not special casing signatures! And that's why the second proposal: https://github.com/shizhMSFT/nv2/pull/19/files#tracking-referrer-metadata

The storgae structure illustrated below is the implementation of docker distribution